groups/promo-infrastructure/alb.tf

```diff
@@ -0,0 +1,258 @@
+      target_type          = "ip"
+      deregistration_delay = 10
+      health_check = {
+        enabled             = true
+        interval            = 30
+        path                = var.health_check_path
+        port                = var.epayments_service_port
+        healthy_threshold   = 3
+        unhealthy_threshold = 3
+        timeout             = 6
+        protocol            = "HTTP"
+        matcher             = "200-399"
+      }
+      tags = {
+        InstanceTargetGroupTag = "${var.application}-epayments-live"
+      }
+    },
+    {
+      name                 = "tg-${var.application}-epayments-test-01"
+      backend_protocol     = "HTTP"
+      backend_port         = var.epayments_service_port
+      target_type          = "ip"
+      deregistration_delay = 10
+      health_check = {
+        enabled             = true
+        interval            = 30
+        path                = var.health_check_path
+        port                = var.epayments_service_port
+        healthy_threshold   = 3
+        unhealthy_threshold = 3
+        timeout             = 6
+        protocol            = "HTTP"
+        matcher             = "200-399"
+      }
+      tags = {
+        InstanceTargetGroupTag = "${var.application}-epayments-test"
+      }
+    }
+  ]
+
+  tags = merge(
+    local.default_tags,
+    map(
+      "ServiceTeam", "${upper(var.application)}-Web-Support"
+    )
+  )
+}
+
+#--------------------------------------------
+# ALB CloudWatch Merics
+#--------------------------------------------
+module "alb_proxy_metrics" {
+  source = "git@github.com:companieshouse/terraform-modules//aws/alb-metrics?ref=tags/1.0.26"
+
+  load_balancer_id = module.promo_proxy_alb.this_lb_id
+  target_group_ids = module.promo_proxy_alb.target_group_arns
+
+  depends_on = [module.promo_proxy_alb]
+}
+
+#--------------------------------------------
+# ALB Target Group Attachments
+#--------------------------------------------
+resource "aws_lb_target_group_attachment" "ebilling" {
+  target_group_arn = [for group in module.promo_proxy_alb.target_group_arns : group if can(regex("ebilling", group))]
+  target_id        = var.ebilling_server_ip
+  port             = var.ebilling_service_port
+
+  depends_on = [
+    module.promo_proxy_alb
+  ]
+}
+
+resource "aws_lb_target_group_attachment" "epayments_live" {
+  target_group_arn = [for group in module.promo_proxy_alb.target_group_arns : group if can(regex("epayments-live", group))]
+  target_id        = var.epayments_live_server_ip
+  port             = var.epayments_service_port
+
+  depends_on = [
+    module.promo_proxy_alb
+  ]
+}
+
+resource "aws_lb_target_group_attachment" "epayments-test" {
+  target_group_arn = [for group in module.promo_proxy_alb.target_group_arns : group if can(regex("epayments-test", group))]
+  target_id        = var.epayments_test_server_ip
+  port             = var.epayments_service_port
+
+  depends_on = [
+    module.promo_proxy_alb
+  ]
+}
```

Comment:
You have 4 target groups but only 3 attachments, is that correct?

Reply:
Yes - 4 target groups and 3 attachments.  The attachment which searches on epayments_live regex is planned to pick up the target groups tg-promo-epayments-live-fwk-01 and tg-promo-epayments-live-01, which uses the same target IP address.

Reply:
For example:

<Location /payments-framework/payments-live/notification>
    ProxyPass        http://ch-lfp-pl1.internal.ch:9000/payments-framework/payments/notification/
    ProxyPassReverse http://ch-lfp-pl1.internal.ch:9000/payments-framework/payments/notification/
  </Location>  <Location /payments-live/payments-framework/payments/smartpay/notification/>
    ProxyPass        http://chf-lfp-pl1.internal.ch:9000/payments-framework/payments/smartpay/notification/
    ProxyPassReverse http://chf-lfp-pl1.internal.ch:9000/payments-framework/payments/smartpay/notification/
  </Location>